src/main/kotlin/org/opensearch/replication/util/ValidationUtil.kt

```diff
@@ -31,6 +31,27 @@ object ValidationUtil {
 
     private val log = LogManager.getLogger(ValidationUtil::class.java)
 
+    fun validateIndexSettings(
+        environment: Environment,
+        followerIndex: String,
+        leaderSettings: Settings,
+        overriddenSettings: Settings,
+        metadataCreateIndexService: MetadataCreateIndexService
+    ) {
+        val settingsList = arrayOf(leaderSettings, overriddenSettings)
```

Comment:
Please validate to see if the behavior is as expected, due to open bug on replication side: https://github.com/opensearch-project/cross-cluster-replication/issues/298?

Reply:
I have fixed this bug in this PR as well. Now we are retrieving the default settings as well .